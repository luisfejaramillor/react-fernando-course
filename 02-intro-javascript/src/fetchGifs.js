import { randomUrl } from "./constants/apiKey"

export const getRandomGif = async ()=> {
    try {
        const result = await fetch(`${randomUrl}`)

        if(!result.ok){
            throw new Error(`Failed to fetch: ${result.status}`);
        }

        const obj = await result.json();
        console.log(obj)

    } catch (error) {
        console.error('Error fetching BREAKING BAD QUOTES:', error);
    }


}