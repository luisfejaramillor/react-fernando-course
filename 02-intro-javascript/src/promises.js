// import { getHeroById } from "./getHeroById";

// export const getHeroByIdAsync = (id) => {
//     return(
//          new Promise((resolve, reject) => {
//             setTimeout(() => {
//                 const hero = getHeroById(id);
//                 hero ? resolve(hero) : reject(`Could not found hero with id: ${id}`)
//             }, 2000);
//         })
//     )
    
// }
