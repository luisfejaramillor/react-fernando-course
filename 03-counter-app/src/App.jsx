function App() {

  return (
    <>
      <h1>HOli</h1>
    </>
  )
}

export default App