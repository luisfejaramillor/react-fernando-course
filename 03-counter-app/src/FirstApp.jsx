import PropTypes from 'prop-types'

export const FirstApp = ({title, subTitle})=> {
    return(
        <>
            <h1 data-testid="test-title" >{title}</h1>
            <p>Primer componente</p>
            <p>{subTitle}</p>
        </>

    )
}


FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
    subTitle: 'No hay subtitulo'
}