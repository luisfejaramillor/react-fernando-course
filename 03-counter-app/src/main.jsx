import ReactDOM from 'react-dom/client'
import './main.css'
import { CounterApp } from './CounterApp'


ReactDOM.createRoot(document.getElementById('root')).render(
        <CounterApp/>
)       