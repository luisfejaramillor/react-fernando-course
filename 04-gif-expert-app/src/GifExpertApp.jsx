import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["Breaking bad"]);

  const handleNewCategory = (category) => {
    if (categories.includes(category)) return;
    setCategories([category, ...categories]);
  };
  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory onNewCategory={handleNewCategory} />
      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};
