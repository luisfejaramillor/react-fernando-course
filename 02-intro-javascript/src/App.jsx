import './App.css'
import { getRandomGif } from './fetchGifs'


getRandomGif()

function App() {

  return (
    <>
      <h1>HOli</h1>
    </>
  )
}

export default App
