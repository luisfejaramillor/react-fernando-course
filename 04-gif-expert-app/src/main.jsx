import ReactDOM from 'react-dom/client'
import { GifExpertApp } from './GifExpertApp'

import './styles/main.css'

ReactDOM.createRoot(document.getElementById('root')).render(
    <GifExpertApp />
)
