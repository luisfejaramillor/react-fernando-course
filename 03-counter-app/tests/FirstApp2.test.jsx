import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Test FirstApp component', () => { 
    const title = 'Soy un titulo'
    const subTitle = 'Soy un SUbtitulo'
    test('Should match the snapshot', () => { 

        const {container} = render(<FirstApp title={title} subTitle={subTitle} />)
        expect(container).toMatchSnapshot()
    })

    test('Must have the message of title', () => {  
        render( <FirstApp title={title} subTitle={subTitle} /> )
        expect(screen.getByText(title)).toBeTruthy()
    })

    test('Must show title in a h1 tag', () => {  
        render( <FirstApp  title={title} subTitle={subTitle} /> )
        expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title)
    })

    test('Debe enviar el subtitulo enviado por props', () => { 
        render( <FirstApp title="" subTitle={subTitle} /> )
        expect(screen.getByText(subTitle)).toBeTruthy()
    })
})