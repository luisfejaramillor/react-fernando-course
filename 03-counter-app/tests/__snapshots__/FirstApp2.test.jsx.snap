// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FirstApp component Should match the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Soy un titulo
  </h1>
  <p>
    Primer componente
  </p>
  <p>
    Soy un SUbtitulo
  </p>
</div>
`;
